src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let switcher = true;
+
+    while (switcher) {
+      switcher = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const firstValue = this[i];
+        const secondValue = this[i + 1];
+        const compareString = (a, b) => (String(a) > String(b) ? 1 : -1);
+
+        const callback =
+          compareFunction === undefined
+            ? compareString(firstValue, secondValue)
+            : compareFunction(firstValue, secondValue);
+
+        if (callback > 0) {
+          this[i] = secondValue;
+          this[i + 1] = firstValue;
+          switcher = true;
+          break;
+        }
+      }
+    }
```

Comment:
The implementation of the sort algorithm here is a bubble sort, which is known to be less efficient for larger datasets. If performance is a concern, consider implementing a more efficient sorting algorithm like quicksort or mergesort.